File: app/src/main/java/id/ac/uasbungaa/helper/MappingHelper.java
package id.ac.uasbungaa.helper;

import android.database.Cursor;

import id.ac.uasbungaa.data.model.Barang;
import id.ac.uasbungaa.data.model.User;

import java.util.ArrayList;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class MappingHelper {
    @NotNull
    public static final MappingHelper INSTANCE;

    @NotNull
    public final User mapCursor(@Nullable Cursor userCursor) {
        User user = new User();
        if (userCursor != null && userCursor.moveToFirst()) {
            do {
                int id = userCursor.getInt(userCursor.getColumnIndexOrThrow("_id"));
                String username = userCursor.getString(userCursor.getColumnIndexOrThrow("username"));
                String password = userCursor.getString(userCursor.getColumnIndexOrThrow("password"));
                String thumbnail = userCursor.getString(userCursor.getColumnIndexOrThrow("thumbnail"));
                user = new User(id, username, password, thumbnail);
            } while (userCursor.moveToNext());
        }

        return user;
    }

    @NotNull
    public final ArrayList mapCursorAllUser(@Nullable Cursor userCursor) {
        ArrayList allUser = new ArrayList();
        if (userCursor != null && userCursor.moveToFirst()) {
            do {
                int id = userCursor.getInt(userCursor.getColumnIndexOrThrow("_id"));
                String username = userCursor.getString(userCursor.getColumnIndexOrThrow("username"));
                String password = userCursor.getString(userCursor.getColumnIndexOrThrow("password"));
                String thumbnail = userCursor.getString(userCursor.getColumnIndexOrThrow("thumbnail"));
                User user = new User(id, username, password, thumbnail);
                allUser.add(user);
            } while (userCursor.moveToNext());
        }

        return allUser;
    }

    @NotNull
    public final ArrayList mapCursorAllBarang(@Nullable Cursor barangCursor) {
        ArrayList allBarang = new ArrayList();
        if (barangCursor != null && barangCursor.moveToFirst()) {
            do {
                int id = barangCursor.getInt(barangCursor.getColumnIndexOrThrow("_id"));
                String nama = barangCursor.getString(barangCursor.getColumnIndexOrThrow("nama"));
                String kode = barangCursor.getString(barangCursor.getColumnIndexOrThrow("kode"));
                int stok = barangCursor.getInt(barangCursor.getColumnIndexOrThrow("stok"));
                int harga = barangCursor.getInt(barangCursor.getColumnIndexOrThrow("harga"));
                String gambar = barangCursor.getString(barangCursor.getColumnIndexOrThrow("gambar"));
                Barang barang = new Barang(id, nama, kode, stok, harga, gambar);
                allBarang.add(barang);
            } while (barangCursor.moveToNext());
        }

        return allBarang;
    }

    @NotNull
    public final ArrayList mapCursorAllBarangKode(@Nullable Cursor barangCursor) {
        ArrayList allBarangKode = new ArrayList();
        if (barangCursor != null && barangCursor.moveToFirst()) {
            do {
                String kode = barangCursor.getString(barangCursor.getColumnIndexOrThrow("kode"));
                allBarangKode.add(kode);
            } while (barangCursor.moveToNext());
        }

        return allBarangKode;
    }

    @NotNull
    public final Barang mapCursorBarang(@Nullable Cursor barangCursor) {
        Barang barang = new Barang();
        if (barangCursor != null && barangCursor.moveToFirst()) {
            do {
                int id = barangCursor.getInt(barangCursor.getColumnIndexOrThrow("_id"));
                String nama = barangCursor.getString(barangCursor.getColumnIndexOrThrow("nama"));
                String kode = barangCursor.getString(barangCursor.getColumnIndexOrThrow("kode"));
                int stok = barangCursor.getInt(barangCursor.getColumnIndexOrThrow("stok"));
                int harga = barangCursor.getInt(barangCursor.getColumnIndexOrThrow("harga"));
                String gambar = barangCursor.getString(barangCursor.getColumnIndexOrThrow("gambar"));
                barang = new Barang(id, nama, kode, stok, harga, gambar);
            } while (barangCursor.moveToNext());
        }

        return barang;
    }

    private MappingHelper() {
    }

    static {
        INSTANCE = new MappingHelper();
    }
}

File: app/src/main/java/id/ac/uasbunga/helper/MappingHelper.kt
package id.ac.uasbunga.helper

import android.database.Cursor
import id.ac.uasbunga.data.local.DatabaseContract.BARANG_200020016.Companion.GAMBAR
import id.ac.uasbunga.data.local.DatabaseContract.BARANG_200020016.Companion.HARGA
import id.ac.uasbunga.data.local.DatabaseContract.BARANG_200020016.Companion.KODE
import id.ac.uasbunga.data.local.DatabaseContract.BARANG_200020016.Companion.NAMA
import id.ac.uasbunga.data.local.DatabaseContract.BARANG_200020016.Companion.STOK
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion.PASSWORD
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion.THUMBNAIL
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion.USERNAME
import id.ac.uasbunga.data.local.DatabaseContract.USER_200020016.Companion._ID
import id.ac.uasbunga.data.model.Barang
import id.ac.uasbunga.data.model.User

object MappingHelper {
    fun mapCursor(userCursor: Cursor?): User {
        var user = User()

        if (userCursor != null && userCursor.moveToFirst()) {
            do {
                val id = userCursor.getInt(userCursor.getColumnIndexOrThrow(_ID))
                val username = userCursor.getString(userCursor.getColumnIndexOrThrow(USERNAME))
                val password = userCursor.getString(userCursor.getColumnIndexOrThrow(PASSWORD))
                val thumbnail = userCursor.getString(userCursor.getColumnIndexOrThrow(THUMBNAIL))
                user = User(id, username, password, thumbnail)
            } while (userCursor.moveToNext())
        }

        return user
    }

    fun mapCursorAllUser(userCursor: Cursor?): ArrayList<User> {
        val allUser = ArrayList<User>()

        if (userCursor != null && userCursor.moveToFirst()) {
            do {
                val id = userCursor.getInt(userCursor.getColumnIndexOrThrow(_ID))
                val username = userCursor.getString(userCursor.getColumnIndexOrThrow(USERNAME))
                val password = userCursor.getString(userCursor.getColumnIndexOrThrow(PASSWORD))
                val thumbnail = userCursor.getString(userCursor.getColumnIndexOrThrow(THUMBNAIL))
                val user = User(id, username, password, thumbnail)
                allUser.add(user)
            } while (userCursor.moveToNext())
        }

        return allUser
    }

    fun mapCursorAllBarang(barangCursor: Cursor?): ArrayList<Barang> {
        val allBarang = ArrayList<Barang>()

        if (barangCursor != null && barangCursor.moveToFirst()) {
            do {
                val id = barangCursor.getInt(barangCursor.getColumnIndexOrThrow(_ID))
                val nama = barangCursor.getString(barangCursor.getColumnIndexOrThrow(NAMA))
                val kode = barangCursor.getString(barangCursor.getColumnIndexOrThrow(KODE))
                val stok = barangCursor.getInt(barangCursor.getColumnIndexOrThrow(STOK))
                val harga = barangCursor.getInt(barangCursor.getColumnIndexOrThrow(HARGA))
                val gambar = barangCursor.getString(barangCursor.getColumnIndexOrThrow(GAMBAR))
                val barang = Barang(id, nama, kode, stok, harga, gambar)
                allBarang.add(barang)
            } while (barangCursor.moveToNext())
        }

        return allBarang
    }

    fun mapCursorAllBarangKode(barangCursor: Cursor?): ArrayList<String> {
        val allBarangKode = ArrayList<String>()

        if (barangCursor != null && barangCursor.moveToFirst()) {
            do {
                val kode = barangCursor.getString(barangCursor.getColumnIndexOrThrow(KODE))
                allBarangKode.add(kode)
            } while (barangCursor.moveToNext())
        }

        return allBarangKode
    }

    fun mapCursorBarang(barangCursor: Cursor?): Barang {
        var barang = Barang()

        if (barangCursor != null && barangCursor.moveToFirst()) {
            do {
                val id = barangCursor.getInt(barangCursor.getColumnIndexOrThrow(_ID))
                val nama = barangCursor.getString(barangCursor.getColumnIndexOrThrow(NAMA))
                val kode = barangCursor.getString(barangCursor.getColumnIndexOrThrow(KODE))
                val stok = barangCursor.getInt(barangCursor.getColumnIndexOrThrow(STOK))
                val harga = barangCursor.getInt(barangCursor.getColumnIndexOrThrow(HARGA))
                val gambar = barangCursor.getString(barangCursor.getColumnIndexOrThrow(GAMBAR))
                barang = Barang(id, nama, kode, stok, harga, gambar)
            } while (barangCursor.moveToNext())
        }

        return barang
    }
}